LLM/rule_book/graph.py

```
@@ -0,0 +1,109 @@
+    answer: str
+    error: Optional[str]
+
+
+async def retrieve(state: RuleState) -> RuleState:
+    index = get_index()
+    if not index._built:
+        return {**state, "chunks": [], "error": "인덱스가 아직 준비되지 않았습니다."}
+    try:
+        chunks = index.search(state["query"], top_k=TOP_K)
+        return {**state, "chunks": chunks, "error": None}
+    except Exception as e:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)